config/config_mgmt.py

```diff
@@ -0,0 +1,709 @@
+#!/usr/bin/env python
+#
+# config_mgmt.py
+# Provides a class for configuration validation and for Dynamic Port Breakout.
+
+try:
+
+    # import from sonic-cfggen, re use this
+    from imp import load_source
+    load_source('sonic_cfggen', '/usr/local/bin/sonic-cfggen')
+    from sonic_cfggen import deep_update, FormatConverter, sort_data
+    from swsssdk import ConfigDBConnector, SonicV2Connector, port_util
+    from json import load
+    from os import system
+    from time import sleep as tsleep
+
+    import sonic_yang
+    import syslog
+    import re
+
+except ImportError as e:
+    raise ImportError("%s - required module not found" % str(e))
+
+# Globals
+# This class may not need to know about YANG_DIR ?, sonic_yang shd use
+# default dir.
+YANG_DIR = "/usr/local/yang-models"
+CONFIG_DB_JSON_FILE = '/etc/sonic/confib_db.json'
+# TODO: Find a place for it on sonic switch.
+DEFAULT_CONFIG_DB_JSON_FILE = '/etc/sonic/default_config_db.json'
+
+# Class to handle config managment for SONIC, this class will use PLY to verify
+# config for the commands which are capable of change in config DB.
+
+class ConfigMgmt():
+
+    def __init__(self, source="configDB", debug=False, allowTablesWithOutYang=True):
+
+        try:
+            self.configdbJsonIn = None
+            self.configdbJsonOut = None
+            self.allowTablesWithOutYang = allowTablesWithOutYang
+
+            # logging vars
+            self.SYSLOG_IDENTIFIER = "ConfigMgmt"
+            self.DEBUG = debug
+
+            self.sy = sonic_yang.SonicYang(YANG_DIR, debug=debug)
+            # load yang models
+            self.sy.loadYangModel()
+            # load jIn from config DB or from config DB json file.
+            if source.lower() == 'configdb':
+                self.readConfigDB()
+            # treat any other source as file input
+            else:
+                self.readConfigDBJson(source)
+            # this will crop config, xlate and load.
+            self.sy.loadData(self.configdbJsonIn)
+
+            # Raise if tables without YANG models are not allowed but exist.
+            if not allowTablesWithOutYang and len(self.sy.tablesWithOutYang):
+                raise Exception('Config has tables without YANG models')
+
+        except Exception as e:
+            print(e)
+            raise(Exception('ConfigMgmt Class creation failed'))
+
+        return
+
+    def __del__(self):
+        pass
+
+    """
+    Return tables loaded in config for which YANG model does not exist.
+    """
+    def tablesWithOutYang(self):
```

Comment:
"Without" is one word, so suggest to rename, `tablesWithoutYang`

Reply:
okies, can be done